### PR TITLE
Add thumbnail images to VSI

### PR DESCRIPTION
**Pull request by johnpryan**
Generates thumbnails using `package:image` This improves the total page load from 41.7 MB to 7.6 MB

### CODE REVIEW

File: web/samples_index/tool/grind.dart [hunk 1]
```diff
@@ -82,6 +83,36 @@ Future scrapeCookbook() async {
   }
 }
 
+@Task('creates thumbnail images in web/images')
+Future createThumbnails() async {
+  await _createThumbnails(Directory('web/images'));
+  await _createThumbnails(Directory('web/images/cookbook'));
+}
+
+// Creates a thumbnail image where each png file
```
Comment (RedBrogdon) [inline web/samples_index/tool/grind.dart:92]: nit: "for each png file?"

File: web/samples_index/tool/grind.dart [hunk 2]
```diff
@@ -82,6 +83,36 @@ Future scrapeCookbook() async {
...
+Future _createThumbnails(Directory directory) async {
+  var files = await directory.list().toList();
+  var filesToWrite = <Future>{};
+
+  for (var entity in files) {
+    var extension = path.extension(entity.path);
```
Comment (RedBrogdon) [inline web/samples_index/tool/grind.dart:98]: Any idea why GitHub is highlighting `extension` as if it were a reserved word? The language tour on dart.dev doesn't list it as one. 🤔

Comment (johnpryan) [reply web/samples_index/tool/grind.dart:98]: It's a keyword as of [2.7](https://dart.dev/guides/language/extension-methods). It's been added to the [grammar](https://github.com/dart-lang/dart-syntax-highlight/blob/aaf98b2af35bb1101ffefc4c9c9a5a3c97be2dbb/grammars/dart.json#L260) for [linguist](https://github.com/github/linguist/tree/master/vendor/grammars), GitHub's highlighter. Creating an issue to add it to the language tour.

Comment (johnpryan) [reply web/samples_index/tool/grind.dart:98]: https://github.com/dart-lang/site-www/pull/2358

**Pull request merged**